src/persist-client/src/internal/maintenance.rs

```diff
@@ -233,7 +233,9 @@ where
 
         if let Some(compactor) = compactor {
             for req in self.compaction {
-                if let Some(handle) = compactor.compact_and_apply_background(machine, req) {
+                if let Some(handle) =
+                    compactor.compact_and_apply_background(machine, req, usize::MAX)
```

Comment:
let's actually set some value for the mem bound before merging the initial version of this PR

Reply:
I wasn't sure for this, since if we don't have part sizes, we pessimistically calculate things based on the `blob_target_size` which could account for a large % of memory on the smaller cluster/storage host sizes. but in thinking about it again, that can't be _worse_ than what we're already doing now 😁

Reply:
yeah, i'm inclined during the transition to treat things with no size as 0 given that the bad case (excessive memory usage) is no worse than it is now and the good case (merging many many small runs into 1) is nice to have